api/specs.proto

```diff
@@ -241,3 +241,15 @@ message ClusterSpec {
 	// CAConfig defines cluster-level certificate authority settings.
 	CAConfig ca_config = 6 [(gogoproto.nullable) = false, (gogoproto.customname) = "CAConfig"];
 }
+
+// ContainerAttachmentSpec specifies the allocation information
+// for the container attachment.
+message ContainerAttachmentSpec {
+	Annotations annotations = 1 [(gogoproto.nullable) = false];
+	
+	// Target specifies the network to attach to.
+	string target = 2;
+	
+	// Addresses are the IPv4/IPv6 addresses specified for this attachment.
+	repeated string addresses = 3;
```

Comment:
No changes it would work as they would today, libnetwork makes sure this info is gossiped among nodes in the network.